compiler/src/compile.rs

```diff
@@ -383,8 +385,14 @@ impl Compiler {
         self.compile_name(name, NameUsage::Load)
     }
 
-    fn store_name(&mut self, name: &str) {
-        self.compile_name(name, NameUsage::Store)
+    fn store_name(&mut self, name: &str) -> CompileResult<()> {
+        if name.eq("__debug__") {
```

Comment:
If check only the `Name` of `compile_store`, it seems that there will be a part different from cpython.

In cpython,
```python
>>> import __debug__
  File "<stdin>", line 1
SyntaxError: cannot assign to __debug__
```

In RustPython, If change to check only the `Name` of `compile_store`,
```python
>>>>> import __debug__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named '__debug__'
```

In my opinion, a check is required in all cases of store to name.

Reply:
Hm, indeed. It also doesn't catch the assignment in `try` and class definitions. On the other hand `store` doesn't appear to be used for function calls/definitions (and maybe other places? I haven't really looked into the compiler code), i.e:

```python
def d(__debug__): pass
```

currently compiles just fine. This is probably because these are `Load`ed and assigned to during invocation inside the eval loop. I think the better approach might be a `forbidden_names` function like CPython uses and invokes wherever a forbidden name assignment *might* occur. @youknowone any opinion on this?

Reply:
I don't know well about compiler part. @coolreader18 could give better answer.

In `forbidden_name`, it only checks `Store` case. So I don't think `forbidden_name` gives an answer for the example. I have no idea about the correct approach. Is this related to bytecode incompatibility?

The pattern how `compile_name` seems like `Result` one to me. I don't know if we only need to check it just in store_name or in more places, but I think their return type would be `Result` at the end of the result.

Reply:
What else should I do in PR now?

Reply:
@DimitrisJim @coolreader18 any suggestion? if we don't have any better idea for now, I suggest to merge it for now and fix the other problems later

Reply:
Yup, for now, that sounds good to me. The linked issue should probably stay open though since we can still assign to it in some cases even after this change. (I'll try and add a comment on the issue on where this assignment is still allowed)